wherehows-web/app/components/datasets/health/score-table.ts

```diff
@@ -0,0 +1,182 @@
+
+  /**
+   * Passed in severity filter. This property lives on the dataset-health container component and its
+   * modification is triggered by a click in the graphing component
+   * @type {string}
+   */
+  currentSeverityFilter: string;
+
+  /**
+   * Passed in category filter. This property lives on the dataset-health container component and its
+   * modification is triggered by a click in the graphing component
+   * @type {string}
+   */
+  currentCategoryFilter: string;
+
+  /**
+   * Calculates table data from the passed in information by appending each row with information that helps to
+   * style the table.
+   * @type {ComputedProperty<Array<IRenderedHealthScore>>}
+   */
+  renderedTableData: ComputedProperty<Array<IRenderedHealthScore>> = computed(
+    'tableData',
+    'currentCategoryFilter',
+    'currentSeverityFilter',
+    function(this: DatasetsHealthScoreTable): Array<IRenderedHealthScore> {
+      const { tableData, currentCategoryFilter: categoryFilter, currentSeverityFilter: severityFilter } = getProperties(
+        this,
+        'tableData',
+        'currentCategoryFilter',
+        'currentSeverityFilter'
+      );
+      return tableData.map(healthScore => ({
+        ...healthScore,
+        isHidden:
+          (!!categoryFilter && healthScore.category !== categoryFilter) ||
+          (!!severityFilter && healthScore.severity !== severityFilter),
+        highlightClass: `dataset-health__score-table__row--${(healthScore.severity || 'normal').toLowerCase()}`
+      }));
+    }
+  );
+
+  /**
+   * Passed in function from the dataset health container that helps us handle when the user clicks on a
+   * specific category from the dropdown menu in the table header
+   * @param {IDropDownOption} selection - the selected option from the dropdown
+   */
+  onCategorySelect: (selection: IDropDownOption<string>) => void;
+
+  /**
+   * Passed in function from the dataset health container that helps us handle when the user clicks on a
+   * specific severity from the dropdown menu on the table header
+   * @param {IDropDownOption} selection - the selected option from the dropdown
+   */
+  onSeveritySelect: (section: IDropDownOption<string>) => void;
+
+  constructor() {
+    super(...arguments);
+
+    setProperties(this, {
+      tableData: this.tableData || [],
+      currentSeverityFilter: this.currentSeverityFilter || '',
+      currentCategoryFilter: this.currentCategoryFilter || '',
+      onCategorySelect: this.onCategorySelect || noop,
+      onSeveritySelect: this.onSeveritySelect || noop
+    });
+  }
+
+  /**
+   * This is attached to each header object so that we can pass into the dropdown option for that header
+   * and allow the user to select from it with a handler
+   * @param type - lets us know which dropdown called this function
+   * @param selection - the actual dropdown option passed in from the child component
+   */
+  onDropdownSelect(type: string, selection: IDropDownOption<string>) {
```

Comment:
Would it make sense for type to be an string enum (or literal, preferably enum), since only specific values are expected?
```typescript
enum HealthScoreDropdownTypeOrSomeBetterDescription {
  Category = 'Category',
  Severity = 'Severity'
}
```
```typescript
if (type === HealthScoreDropdownTypeOrSomeBetterDescription.Category ) {
  ...
}
if (type === HealthScoreDropdownTypeOrSomeBetterDescription.Severity ) {
 ...
}
```